### PR TITLE
[sql lab] deeper support for templating

### PR DESCRIPTION
**Pull request by mistercrunch**
params

### CODE REVIEW

**Comment by john-bodley** [comment]
cc: @graceguo-supercat @michellethomas @williaster

**Review by graceguo-supercat — APPROVED** [review]
LGTM

File: superset/views/core.py [hunk 1]
```diff
@@ -2161,6 +2161,8 @@ def sql_json(self):
         sql = request.form.get('sql')
         database_id = request.form.get('database_id')
         schema = request.form.get('schema') or None
+        template_params = json.loads(
+            request.form.get('templateParams') or '{}')
```
Comment (betodealmeida) [inline superset/views/core.py]: `get` takes a default parameter:

```python
template_params = json.loads(request.form.get('templateParams', '{}'))
```

File: superset/assets/javascripts/SqlLab/components/TemplateParamsEditor.jsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    if (isValid) {
+      this.props.onChange(codeText);
+    } else {
+      this.props.onChange('{}');
+    }
+  }
+  renderDoc() {
+    return (
+      <p>
+        Assign a set of parameters as <code>JSON</code> bellow
```
Comment (betodealmeida) [inline superset/assets/javascripts/SqlLab/components/TemplateParamsEditor.jsx]: s/bellow/below/g

**Review by betodealmeida — APPROVED** [review]
Looks good to me, I just have a couple comments.

File: superset/views/core.py (hunk 1 above)
Comment (mistercrunch) [reply superset/views/core.py]: The intent of the `or` is to handle an empty string, though I could probably trim

**Pull request merged**